src/DistributedLock.AzureStorage/AzureStorageDistributedLock.cs

```diff
@@ -274,66 +275,62 @@ private async Task<bool> AcquireLeaseAsync(AzureBlobLease lease)
             try
             {
                 string newToken;
-                await leaseBlob.FetchAttributesAsync(null, null, _storageContext).ConfigureAwait(false);
+                var props = await leaseBlob.GetPropertiesAsync().ConfigureAwait(false);
 
-                if (leaseBlob.Properties.LeaseState == LeaseState.Leased)
+                if (props.Value.LeaseState == LeaseState.Leased)
                 {
                     if (string.IsNullOrEmpty(lease.Token))
                     {
                         return false;
                     }
 
-                    newToken = await leaseBlob.ChangeLeaseAsync(
-                        newLeaseId,
-                        AccessCondition.GenerateLeaseCondition(lease.Token),
-                        null,
-                        _storageContext).ConfigureAwait(false);
+                    var response = await leaseBlob.GetBlobLeaseClient(lease.Token).ChangeAsync(newLeaseId);
+                    newToken = response.Value.LeaseId;
                 }
                 else
                 {
                     try
                     {
-                        newToken = await leaseBlob.AcquireLeaseAsync(
-                                Options.LeaseDuration,
-                                newLeaseId,
-                                null,
-                                null,
-                                _storageContext)
-                            .ConfigureAwait(false);
+                        var response = await leaseBlob.GetBlobLeaseClient(lease.Token).AcquireAsync(Options.LeaseDuration).ConfigureAwait(false);
+                        newToken = response.Value.LeaseId;
                     }
-                    catch (StorageException se)
-                        when (se.RequestInformation != null
-                              && se.RequestInformation.ErrorCode.Equals(BlobErrorCodeStrings.LeaseAlreadyPresent, StringComparison.OrdinalIgnoreCase))
+                    catch (RequestFailedException se)
+                        when (se != null
+                              && se.ErrorCode == BlobErrorCode.LeaseAlreadyPresent)
                     {
                         return false;
                     }
                 }
 
                 lease.Token = newToken;
                 lease.IncrementEpoch(); // Increment epoch each time lease is acquired or stolen by a new host
-                await leaseBlob.UploadTextAsync(
-                    JsonConvert.SerializeObject(lease),
-                    null,
-                    AccessCondition.GenerateLeaseCondition(lease.Token),
-                    null,
-                    _storageContext).ConfigureAwait(false);
+                using (var uploadFileStream = new MemoryStream(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(lease) ?? "")))
```

Comment:
@Master-Oliveira we were using before leaseBlob.UploadTextAsync what about create a SetBlobState private method to re use the way we upload it with the memory stream ?

Reply:
Great idea, done ;)